cdap-app-fabric/src/main/java/io/cdap/cdap/master/environment/k8s/MasterEnvironmentMain.java

```diff
@@ -138,4 +149,35 @@ public static void doMain(String[] args) throws Exception {
       shutdownLatch.countDown();
     }
   }
+
+  /**
+   * Return {@link InternalAuthenticator} with
+   *   {@link SystemAuthenticationContext} if cdap-secret is mounted (e.g. when only running system code / trusted code)
+   *   or {@link WorkerAuthenticationContext} if cdap-secret is not mounted (e.g. running untrusted user provided code)
+   */
+  private static InternalAuthenticator getInternalAuthenticator(CConfiguration cConf) {
+    File sConfFile = new File(cConf.get(Constants.Twill.Security.MASTER_SECRET_DISK_PATH));
+    Injector injector;
+    if (sConfFile.exists()) {
```

Comment:
What if MASTER_SECRET_DISK_PATH has been set, but the path does not exist? Shouldn't we throw an exception instead of executing the else block?

Reply:
The condition below is based on if path exists or not, not based on if the config is set.

Reply:
Correct. But if one makes a mistake in setting the path?